File: app/src/main/java/cn/nicolite/huthelper/listener/ActivityLifeCycleListener.java
package cn.nicolite.huthelper.listener;

import android.os.Bundle;

/**
 * Activity生命周期监听
 * Created by nicolite on 17-10-14.
 */

public interface ActivityLifeCycleListener {

    void onCreate(Bundle saveInstanceState);

    void onStart();

    void onResume();

    void onPause();

    void onStop();

    void onDestroy();

    void onRestart();
}

File: app/src/main/java/cn/nicolite/huthelper/listener/ActivityLifeCycleListener.kt
package cn.nicolite.huthelper.listener

import android.os.Bundle

/**
 * Activity生命周期监听
 * Created by nicolite on 17-10-14.
 */

interface ActivityLifeCycleListener {

    fun onCreate(saveInstanceState: Bundle?)

    fun onStart()

    fun onResume()

    fun onPause()

    fun onStop()

    fun onDestroy()

    fun onRestart()
}
